src/sentry/native/native_sdk.cpp

```diff
@@ -0,0 +1,207 @@
+		if (!frame.context_line.is_empty()) {
+			sentry_value_set_by_key(sentry_frame, "context_line", sentry_value_new_string(frame.context_line.utf8()));
+			sentry_value_set_by_key(sentry_frame, "pre_context", sentry::native::strings_to_sentry_list(frame.pre_context));
+			sentry_value_set_by_key(sentry_frame, "post_context", sentry::native::strings_to_sentry_list(frame.post_context));
+		}
+		sentry_value_append(frames, sentry_frame);
+	}
+
+	sentry_value_set_by_key(stack_trace, "frames", frames);
+	sentry_value_set_by_key(exception, "stacktrace", stack_trace);
+	sentry_event_add_exception(event, exception);
+	sentry_capture_event(event);
+}
+
+void NativeSDK::initialize() {
+	ERR_FAIL_NULL(OS::get_singleton());
+	ERR_FAIL_NULL(ProjectSettings::get_singleton());
+
+	sentry_options_t *options = sentry_options_new();
+	sentry_options_set_dsn(options, SentryOptions::get_singleton()->get_dsn());
+
+	// Establish handler path.
+	String handler_fn;
+#ifdef LINUX_ENABLED
+	handler_fn = "crashpad_handler";
+#elif WINDOWS_ENABLED
+	handler_fn = "crashpad_handler.exe";
+#elif MACOS_ENABLED
+	handler_fn = "crashpad_handler";
+#endif
+	String handler_path = OS::get_singleton()->get_executable_path().get_base_dir() + "/" + handler_fn;
+	if (!FileAccess::file_exists(handler_path)) {
+		handler_path = ProjectSettings::get_singleton()->globalize_path("res://addons/sentrysdk/bin/" + handler_fn);
+	}
+	if (FileAccess::file_exists(handler_path)) {
+		sentry_options_set_handler_path(options, handler_path.utf8());
+	} else {
+		ERR_PRINT(vformat("Sentry: Failed to locate crash handler (crashpad) - backend disabled (%s)", handler_path));
+		sentry_options_set_backend(options, NULL);
+	}
+
+	sentry_options_set_database_path(options, (OS::get_singleton()->get_user_data_dir() + "/sentry").utf8());
+	sentry_options_set_sample_rate(options, SentryOptions::get_singleton()->get_sample_rate());
+	sentry_options_set_release(options, SentryOptions::get_singleton()->get_release());
+	sentry_options_set_debug(options, SentryOptions::get_singleton()->is_debug_enabled());
+	sentry_options_set_environment(options, sentry::environment::get_environment());
+	sentry_options_set_sdk_name(options, "sentry.native.godot");
+	sentry_options_set_max_breadcrumbs(options, SentryOptions::get_singleton()->get_max_breadcrumbs());
+
+	// Attach LOG file.
+	// TODO: Decide whether log-file must be trimmed before send.
+	if (SentryOptions::get_singleton()->is_attach_log_enabled()) {
+		String log_path = ProjectSettings::get_singleton()->get_setting("debug/file_logging/log_path");
+		if (FileAccess::file_exists(log_path)) {
+			log_path = log_path.replace("user://", OS::get_singleton()->get_user_data_dir() + "/");
+			sentry_options_add_attachment(options, log_path.utf8());
+		} else {
+			WARN_PRINT("Sentry: Log file not found. Make sure \"debug/file_logging/enable_file_logging\" is turned ON in the Project Settings.");
```

Comment:
Just saw below `print_debug` and `print_error`. Should we align on a single logging strategy here (ideally with configurable verbosity akin to diagnostic logger I mentioned above)?

Reply:
We can wrap it in a custom logger if we need one. This is a built-in godot macro. If we want this message to end up reported to the developer in the Godot's **Errors** panel - we should be using it.

`ERR_PRINT` and `WARN_PRINT` are standard idiomatic godot logger macros that wrap also a line, and a file from where this call is made. We can make our own logger too, but this should be taken into consideration, because users *expect* errors to show up in the **Errors** panel and in the app's log file.